### ISSUE TITLE
cornerPadding not working?

### ISSUE THREAD

**Issue by johanndms** [pixolith/fabricjs-customise-controls-extension#8]
I'm trying to set an offset for the controls, using cornerPadding but this doesn't seem to work? Any suggestions?

I also notice that using the "mtr" rotate handle at the top has a weird reaction on the first click....

Thanks
J

`<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title></title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.0/jquery.min.js"></script>
    <script src="http://cdnjs.cloudflare.com/ajax/libs/fabric.js/1.5.0/fabric.min.js"></script>
    <script src="js/customiseControls.js"></script>

</head>
<body>
<canvas id="mycanvas"></canvas>
</body>

<script>
    $(function () {

        fabric.Object.prototype.setControlsVisibility( {
            mt: false
        });

        fabric.Canvas.prototype.customiseControls( {
            tl: {
                action: 'scale'
            },
            tr: {
                action: 'scale'
            },
            bl: {
                action: 'scale'
            },
            br: {
                action: 'scale'
            },
            mr: {
                action: 'moveUp'
            },
            ml: {
                action: 'moveDown'
            },
            mb: {
                action: 'remove'
            },
            mtr: {
                action: 'rotate'
            }
        } );

        // basic settings
        fabric.Object.prototype.customiseCornerIcons( {
            settings: {
                borderColor: '#cccccc',
                cornerSize: 25,
                cornerShape: 'circle',
                cornerBackgroundColor: '#cccccc',
                cornerPadding: 20 // Not working
            },
            tr: {
                icon: 'js/controls/resize-tr.svg'
            },
            tl: {
                icon: 'js/controls/resize-tl.svg'
            },
            br: {
                icon: 'js/controls/resize-tl.svg'
            },
            bl: {
                icon: 'js/controls/resize-tr.svg'
            },
            mtr: {
                icon: 'js/controls/rotate.svg'
            },
            mb: {
                icon: 'js/controls/remove.svg'
            },
            ml: {
                icon: 'js/controls/down.svg'
            },
            mr: {
                icon: 'js/controls/up.svg'
            }

        } );

        canvas = new fabric.Canvas('mycanvas', {width: 1200, height: 900});

        var rect = new fabric.Rect({
            width: 300,
            height: 300,
            top: 100,
            left: 100,
            fill: 'rgba(255,0,0,0.5)'
        });

        canvas.add(rect);

    });

</script>

</html>`

**Comment by MDSLKTR** [comment]
Hello,

First of all thank you very much for reporting this.
Maybe there is a missunderstanding in what cornerPadding really is. It's the inner distance between the icon and the corner box borders. I made some screens with the difference, where the first has a padding of 10 and second 15.
![cp10](https://cloud.githubusercontent.com/assets/9129687/13438691/632522bc-dfea-11e5-83a4-b4df0ad7653f.jpg)
![cp15](https://cloud.githubusercontent.com/assets/9129687/13438692/632b6c3a-dfea-11e5-8f72-f093a10d8753.jpg)

If however it doesn't work in some scenario you have, a code example would be of great help.
Concerning the click on the mtr rotating point, i do not see what something weird on click so far, so maybe you can provide a screenshot so i can see what you mean. Also which version of customizse controls are you using? The one for fabric 1.5.0 or the one for the current master.

Simon

**Comment by johanndms** [comment]
Hi Simon,

Thanks for the quick response and info.

Oops - think I get it. Was mistaking cornerPadding on the controls for setting "padding" on the actual object.

Apologies, your code is correct on the padding :+1: 

With regard to the mtr - still seem to figure that one out.... First click to rotate moves the object upwards?

http://95.168.192.184/dev4/padding.html

https://www.dropbox.com/s/ama9k8hmdb1lbhf/mtr-first-click.mov?dl=0

Thanks
Johann

**Comment by MDSLKTR** [comment]
Hello again,

sorry if the example wasn't clear enough but that's kind of how it should work as you said now. Shifting the controls to a certain offset is not possible afaik, or at least not with huge hacking into the core. So the padding is just for scaling your icons inside the box properly. You can increase the size of the boxes with cornerSize (same as fabric.js standard behaviour).

For the rotating point problem try setting `originX: 'center` and `originY: 'center'` to the object like so in your rect example

`var rect = new fabric.Rect({
            originX: 'center',
            originX: 'center',
           width: 300,
            height: 300,
            top: 100,
            left: 100,
            fill: 'rgba(255,0,0,0.5)'
        });
`

This will rotate the rect around the center instead of the top left corner. It's kind of weird that this is not the default setting but oh well.

Simon

**Comment by johanndms** [comment]
Hello Simon,

Thanks - that works perfect now!!!

For the padding and rotate I now use:
rect.set({
            padding: 12,
            originX: 'center',
            originY: 'center'
        });

and this works perfect!!

Thanks once again for the assistance and plugin!!

Cheers,
Johann
